docs/specifications/1-tenets.md

```diff
@@ -0,0 +1,21 @@
+---
+version: 1.0.0
+---
+# Design Tenets (unless you know better ones)
+
+* *jsii* APIs strive to feel idiomatic in all supported languages.
+* *jsii* applications behave identically regardless of the language they are
+  written in. It favors correctness over performance.
+* *jsii* **does not** attempt to support all TypeScript idioms (many features of
+  TypeScript cannot be expressed in some target languages).
+  * When prohibiting an idiom, *jsii* strives to provide an error message that
+    gives the user insight into why the pattern cannot be supported.
+* *jsii* guides (instead of forcing) library authors towards better APIs:
+  * Reserved names are limited to a minimum.
+  * Warn about patterns or names that are known to be awkward in other
+    programming languages, instead of blocking them.
```

Comment:
I don't think that should be a jsii tenent - jsii should allow users to define any API they desire as long as it uses the supported features.

Reply:
The tenet I want to propose is something along the lines of "don't _force_ opinions on how APIs should be designed", which implies minimizing the things *jsii* prohibits.